### ISSUE TITLE
Interoperability with pthreads

### ISSUE THREAD

**Issue by jprotze** [OpenMPToolsInterface/LLVM-openmp#16]
```
#include <thread>
void f() {
    #pragma omp parallel
    {
    }
}
int main() {
    std::thread t1(f);
    std::thread t2(f);
    t1.join();
    t2.join();
}
```

If a runtime supports above program, the OpenMP runtime should provide a thread-creation callback for each of the C++/pthreads that execute a parallel region. The expectation from the execution model of OpenMP would be that these threads are `initial threads`.

**Comment by omalyshe** [comment]
I'm working on it.

**Comment by jprotze** [comment]
Just pushed test-case in 2915c59

**Comment by jprotze** [comment]
Fixed in PR #17
